frontend/app/lib/validate.server.ts

```diff
@@ -1,12 +1,66 @@
 import { z } from 'zod'
 import { CornerType, PositionType, SlideAnimationType } from './types.js'
+import { isWalletAddress } from './utils.js'
 
 const rangeError = { message: 'Value has to be between 16 and 24' }
 
 export const walletSchema = z.object({
-  walletAddress: z.string().min(1, { message: 'Wallet address is required' })
+  walletAddress: z
+    .string()
+    .min(1, { message: 'Wallet address is required' })
+    .superRefine(async (url, ctx) => {
+      if (url.length === 0) return
+
+      try {
+        const parsed = new URL(toWalletAddressUrl(url))
+        if (parsed.protocol !== 'https:') {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: 'Wallet address must use HTTPS protocol'
+          })
+          return
+        }
+
+        await isValidWalletAddress(parsed.toString())
```

Comment:
I think errors from this one should handled differently. Like we have more detailed errors available (not found etc.), and this isn't always "invalid wallet address format" error.